### ISSUE TITLE
cargo package registry is not casesensitive

### ISSUE THREAD

**Issue by JHenneberg** [go-gitea/gitea#28762]
### Description

I have a crate which i publish using the gitea package registry. Changing the name from Uppercase to Lowercase has no effect for the package registry. The published package is still registred as Uppercase although the name in the cargo.toml has been changed and also the created *.crate file is correct. After publishing there is a test if the crate is avaiable which fails. If I change the create name to any other name the package is release in a correct manner.

### Gitea Version

1.21.3

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Windows 11

### How are you running Gitea?

Docker

### Database

None

**Comment by delvh** [comment]
It seems to be intended.
After all, it wouldn't help if you could publish a package with the same name but different capitalization.
That only leads to confusion in other places.
As you detailed yourself, the workaround
- rename to something else
- rename to new capitalization

should always work for this uncommon usecase.

**Comment by JHenneberg** [comment]
I was reading this in a rust issue somewhere correct, talking about crates.io .

But in this case it is a bug, because i deleted the complete registry and published packages from the organization on gitea before I published the lowercase version and this should be possible and not changed to the previous published one.

**Comment by JHenneberg** [comment]
@KN4CK3R 
Using the GUI from Gitea the package registry is creating an repository.

I created this repository using Giteas package options for rust. After I uploaded a package (`Mypackage`) in uppercase I deleted this repository and also the packages itself from the organization. 
Then I changed the crates name to lowercase (`mypackage`) and setup a new repository using the automatism of Gitea.
I would expect that the freshly published crate would be managed as lowercase now, but it is not.

Publishing other packes using the same registry (repository) the lowercase name is not turned into uppercase. So I expect some caching mechanism which is not resettet to force the crate to be uppercase.

Hope this is easier to understand.
